cmd/state-offline-installer/offline-installer.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/ActiveState/cli/internal/captain"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+)
+
+const (
+	cmdInstall   = "install"
+	cmdUnInstall = "uninstall"
+)
+
+type Params struct {
+	sourcePath      string
+	path            string
+	backpackZipFile string
+}
+
+func newParams() *Params {
+	return &Params{sourcePath: ".", path: "/tmp", backpackZipFile: os.Args[0]}
+}
+
+func main() {
+	if os.Getenv("VERBOSE") == "true" {
+		logging.CurrentHandler().SetVerbose(true)
+	}
+
+	runErr := run()
+	if runErr != nil {
+		errMsg := errs.Join(runErr, ": ").Error()
+		if locale.IsInputError(runErr) {
+			logging.Debug("state-offline-installer errored out due to input: %s", errMsg)
+		} else {
+			logging.Critical("state-offline-installer errored out: %s", errMsg)
+		}
+
+		fmt.Fprintln(os.Stderr, errMsg)
+	}
+}
+
+func run() error {
+	args := os.Args
+
+	out, err := output.New("", &output.Config{
+		OutWriter: os.Stdout,
+		ErrWriter: os.Stderr,
+	})
+	if err != nil {
+		return errs.Wrap(err, "Could not initialize outputer")
+	}
+
+	p := primer.New(nil, out, nil, nil, nil, nil, nil, nil, nil, nil)
```

Comment:
I would strongly suggest using primer the way it was intended. I see you are using config, prompter, analytics, .. throughout your code. Ideally you configure that only once on the primer, and then you pass the primer struct along, rather than initialize it again in your code.

Doing it this way can lead to race conditions. eg. you could initialize config twice which could lead to writes/reads happening in async without awareness of one another.